src/bam/collapse_reads_to_fragments/pipeline.rs

```diff
@@ -188,24 +188,37 @@ impl<W: io::Write> CallConsensusRead<W> {
                                     RecordStorage::SingleRecord { .. } => unreachable!(),
                                 };
                                 let overlap = calc_overlap(&l_rec, &record)?;
-                                if overlap > 0 {
-                                    let uuid = &Uuid::new_v4().to_hyphenated().to_string();
+                                match (
+                                    overlap > 0,
+                                    is_valid_overlap(
+                                        overlap as u32,
+                                        l_rec.cigar_cached().unwrap().into_iter().rev(),
+                                    ),
+                                    is_valid_overlap(
+                                        overlap as u32,
+                                        record.cigar_cached().unwrap().into_iter(),
+                                    ),
+                                ) {
```

Comment:
I'm not sure on how rust is resolving this.
I preferred adding three expressions into the match-clause as this would prevent code duplication.
Still, `is_valid_overlap` only needs to be resolved in case `overlap > 0`.
So, what is rust actually doing in case the first `overlap > 0` is `false`? Directly entrering the `_`-arm or will `is_valid_overlap` be evaluated?

Reply:
Might be that the compiler recognizes this indeed. I think it is fine like this.

Reply:
It does not. Short circuiting is done using `if <cond> && <cond> {} else {}`
`match` is nice, but in this case it's not the right choice.

Reply:
Thanks for pointing out. You totally forget that there are if-closures when writing rust code :D